lib/reversetunnel/agentpool.go

```diff
@@ -255,8 +255,21 @@ func (m *AgentPool) pollAndSyncAgents() {
 	}
 }
 
+// getProxyDetailsFromTheOldestCachedCall gets the cached ProxyDetails obtained during the oldest cached agent.connect call.
+func (m *AgentPool) getProxyDetailsFromTheOldestCachedCall(addr utils.NetAddr) *ProxyDetails {
+	agents, ok := m.agents[addr]
```

Comment:
Should this be accessed under a lock?

Reply:
the addAgent-> getProxyDetailsFromTheOldestCachedCall call is invoked inside `m.withLock(...)` [block](https://github.com/gravitational/teleport/blob/eb7bb01d34a84a253e7a46f51ac53d17c76c8654/lib/reversetunnel/agentpool.go#L196 ) so the access to m.agents is protected by `m.mu`Lock

Reply:
Gotcha. In cases like this I usually either add `Locked` suffix to the function name, or update the godoc to say that this method should only be called under a lock.